src/ui/common/src/reducers/integration.ts

```diff
@@ -154,7 +154,17 @@ export const handleLoadIntegrationObject = createAsyncThunk<
       return thunkAPI.rejectWithValue(objectResponseBody.error);
     } else {
       const serialized = (objectResponseBody as ObjectPreviewResponse).data;
-      return JSON.parse(serialized) as Data;
+      const rawData = JSON.parse(serialized);
+      if ('schema' in rawData) {
+        return rawData as Data;
+      }
+
+      // this is a bson_table
+      const rows = rawData as TableRow[];
+      return {
+        schema: inferSchema(rows),
```

Comment:
Is the reason we need `inferSchema` that bson table doesn't have `schema` field populated?